File: apple/Sources/CCDAEngine/Models/CCDAHumanName.swift
import Foundation

public struct CCDAHumanName: Hashable {
    public let prefix: String?
    public let given: [String]
    public let family: String?

    public var display: String {
        ([prefix] + given + [family]).compactMap { $0 }.joined(separator: " ")
    }
}

File: android/ccda-engine/src/main/kotlin/com/ccdakit/engine/Models/Patient/CCDAHumanName.kt
package com.ccdakit.engine

/**
 * Patient human name parsed from CDA name components.
 *
 * @property prefix Name prefix such as title or honorific.
 * @property given Given name components.
 * @property family Family name.
 */
data class CCDAHumanName(
    val prefix: String? = null,
    val given: List<String> = emptyList(),
    val family: String? = null,
)
